### ISSUE TITLE
Toggle buttons on Settings page are not accessible

### ISSUE THREAD

**Issue by bbsmooth** [freeCodeCamp/freeCodeCamp#49214]
### Describe the Issue

I've been wanting to fix the toggle buttons on the Settings page for quite a while now. PR #48567 has already been started to implement a new ToggleButton component. I wanted to create this issue to make sure that any new components address all of the current accessibility issues. The current toggles have several issues:

- Their labels are not associated with their input
- They are not always keyboard accessible
- Some of them do not use the correct inputs for their functionality

There are two types of toggles on the Settings page:

- Toggles that invoke change immediately when the toggle value is changed (e.g. Night Mode, Campfire Mode).
- Toggles that only take effect when the corresponding Save button is clicked (e.g. the Privacy Settings).

These two toggles need to be created a little differently.

One option for the immediately invoked toggles is to use a group of two `button` elements, using the `aria-pressed` attribute to signal to screen readers which one is currently set. I have put together a codepen to demonstrate how this should be marked up and function.

[Accessible Toggle Group Using Buttons](https://codepen.io/bbsmooth/pen/PoBGVOZ)

Some of these types of toggles are of the On/Off variety and thus we could use an aria switch input for them if we wanted to introduce a third type of toggle. But I don't think this is a necessity, especially because one of these toggles is not On/Off. I think it would be better to be consistent across all of these toggles. So either use `button`s for them all or make them all On/Off toggles and use a switch.

The "delayed" toggles that can only be triggered with the Save button should be composed of a group of radio buttons. No special aria attributes need to be used here.

[Accessible Toggle Group Using Radio Button Inputs](https://codepen.io/bbsmooth/pen/KKBNmym)

A few questions:

- Do we want to create separate components for each type of toggle?
- The toggles should be wrapped in `fieldset`/`legend`s. Do we want to include that in these toggles?

### Affected Page

https://www.freecodecamp.org/settings

### Steps to Reproduce

N/A

### Expected behavior

See above write-up.

### System

N/A

**Comment by Sboonny** [comment]
I am in favor of two components that looks different, because they have different interaction with the API.

I am in favor of adding the `legend` and `fieldset`, although styling can be hard, we can add property `all: unset` to them, which put them in the blank state in the canvas, and we can style them however we want.

**Comment by bbsmooth** [comment]
> I am in favor of two components that looks different, because they have different interaction with the API.

I think that would be a good idea since their keyboard interaction is slightly different and this would let visual keyboard users know that.

**Comment by Sboonny** [comment]
- we need to add focus indication for both, which should be sorted in this PR https://github.com/freeCodeCamp/freeCodeCamp/pull/49579

[rec-tab (15).webm](https://user-images.githubusercontent.com/88248797/223756574-7cd0b657-c1d6-4fc4-bd69-049fa6da7534.webm)

- wrap the input in fieldset and legend for accessibility.


---

We can open new issue for the components to keep this changes small.

If you would like to fix this issue, please make sure you read [our guidelines for contributing](https://contribute.freecodecamp.org/#/), we prioritize contributors following the instructions in our guides. Join us in [our chat room](https://discord.gg/PRyKn3Vbay) or [the forum](https://forum.freecodecamp.org/c/contributors/3) if you need help contributing, our prolific contributors will guide you through this.

**Comment by Sboonny** [comment]
To solve this issue, wrap the input in fieldset and legend for accessibility.
